### ISSUE TITLE
No metrics when I use tensorflow estimator

### ISSUE THREAD

**Issue by xzy0223** [aws/aws-step-functions-data-science-sdk-python#40]
hi

I'm trying step functions data science sdk, but I ran into a issue

I created tf estimator in my jupyter  notebook like this:
`tf_estimator = TensorFlow(
                       entry_point='my_train_1.py',
                       model_dir=model_dir,
                       output_path=s3_output_location,
                       train_instance_type=train_instance_type,
                       train_instance_count=1,
                       hyperparameters=hyperparameters,
                       role=sagemaker.get_execution_role(),
                       #base_job_name='tf-scriptmode-mnist',
                       framework_version='2.0.0',
                       py_version='py3',
                       metric_definitions=metric_definitions,
                       script_mode=True)`

I really defined the "metric_definitions", but when I chained all steps as a workflow, and executed it.

there is no customer metrics in sagemaker console

but when I just create the estimator and fit without step functions data science sdk, I can get the metric.

what did I miss? or any suggestion?

thanks a lot

**Comment by hanaan-yousaf** [comment]
I had the same issue using `sagemaker.sklearn.estimator.SKLearn`

I had to manually add them to the step function definition under `AlgorithmSpecification` in the training state like so

```json
"AlgorithmSpecification": {
  "TrainingImage": "someimage",
  "MetricDefinitions": [ 
         { 
            "Name": "foo",
            "Regex": "bar=(.*?);"
         }
      ]
}
```

**Comment by xzy0223** [comment]
> nually add them to the step function definit

thanks, bro!!!

**Comment by shunjd** [comment]
It happens to be an issue in SageMaker SDK, I've created an issue on their repo.
